python/ray/serve/api.py

```diff
@@ -718,3 +719,30 @@ def my_deployment_function(request):
     """
     _request_context = ray.serve.context._serve_request_context.get()
     return _request_context.multiplexed_model_id
+
+
+@PublicAPI(stability="alpha")
+def status() -> ServeStatus:
```

Comment:
When the GCS is down, does this API return an empty status or does it raise an exception?

Reply:
Thats a great question, I'm actually not sure. It will likely raise an exception, but I'm not sure what. Do you know if there's a straightforward way to test it?

Seems like if the client is cached, it will try to call `get_serve_details()` on the controller, which will raise a `RayActorError` since the controller is dead. If the client isn't cached, it will try to get a reference to the controller, I'm not sure if it `get_global_client` will return None in that case or raise an error.

Reply:
Followed up offline.